<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>425</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>425</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Innerspec assignment - Iñigo Marquínez</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/logos/resources/Innerspec.png</normaloff>:/logos/resources/Innerspec.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="1" column="0">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout_main">
      <item row="2" column="0">
       <widget class="QGroupBox" name="groupBox_visualizationFeatures">
        <property name="title">
         <string>Visualization features</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QWidget" name="widget_visualizationFeatures" native="true">
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="0" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout_dataWindowSize">
              <item>
               <widget class="QSpinBox" name="spinBox_dataWindowWidth">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="toolTip">
                 <string>Set the width of the plot window</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_x">
                <property name="minimumSize">
                 <size>
                  <width>20</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>20</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>x</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinBox_dataWindowHeight">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="toolTip">
                 <string>Set the height of the plot window</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="comboBox_lineStyle">
              <property name="toolTip">
               <string>Select the line style of the plot</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_lineStyle">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>200</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Line style</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_dataWindowSize">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>200</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Data window size</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0" colspan="2">
             <widget class="QPushButton" name="pushButton_color">
              <property name="toolTip">
               <string>Select the line color of the plot</string>
              </property>
              <property name="text">
               <string>Select color</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QGroupBox" name="groupBox_advancedFeatures">
        <property name="title">
         <string>Advanced features</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="QWidget" name="widget_advancedFeatures" native="true">
           <layout class="QGridLayout" name="gridLayout_8">
            <item row="0" column="1">
             <widget class="QDoubleSpinBox" name="doubleSpinBox_amplitude">
              <property name="toolTip">
               <string>Set the initial amplitud of the wave</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_amplitude">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>200</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Amplitude</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_attenuationFactor">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>200</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="toolTip">
               <string>Set the attenuation factor (units per second)</string>
              </property>
              <property name="text">
               <string>Attenuation factor</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QDoubleSpinBox" name="doubleSpinBox_attenuationFactor">
              <property name="toolTip">
               <string>Set the attenuation factor (units per second)</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QGroupBox" name="groupBox_waveParameters">
        <property name="title">
         <string>Wave parameters</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_6">
         <item row="0" column="0">
          <widget class="QWidget" name="widget_waveParameters" native="true">
           <layout class="QGridLayout" name="gridLayout_7">
            <property name="sizeConstraint">
             <enum>QLayout::SetNoConstraint</enum>
            </property>
            <item row="0" column="1">
             <widget class="QDoubleSpinBox" name="doubleSpinBox_oscillationFrequency">
              <property name="toolTip">
               <string>Set the oscillation frequency of the wave, in Hertz</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="spinBox_numberOfPeriods">
              <property name="toolTip">
               <string>Set an integer number of periods to be drawn</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_osillationFrequency">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>200</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Oscillation frequency [Hz]</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_numberOfPeriods">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>200</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Number of periods</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QDoubleSpinBox" name="doubleSpinBox_initialDelay">
              <property name="toolTip">
               <string>Set the initial delay of the wave, in degrees</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_initialDelay">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>200</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Initial delay [degrees]</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_samplingFrequency">
              <property name="minimumSize">
               <size>
                <width>200</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>200</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Sampling frequency [Hz]</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QDoubleSpinBox" name="doubleSpinBox_samplingFrequency">
              <property name="toolTip">
               <string>Set the sampling frequency of the wave</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>425</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_saveToFile"/>
   </widget>
   <widget class="QMenu" name="menuShow">
    <property name="title">
     <string>Show</string>
    </property>
    <addaction name="action_advancedFeatures"/>
    <addaction name="action_visualizationFeatures"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuShow"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_saveToFile">
   <property name="text">
    <string>Save to file</string>
   </property>
  </action>
  <action name="action_advancedFeatures">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Advanced features</string>
   </property>
  </action>
  <action name="action_visualizationFeatures">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Visualization features</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>doubleSpinBox_oscillationFrequency</tabstop>
  <tabstop>spinBox_numberOfPeriods</tabstop>
  <tabstop>doubleSpinBox_initialDelay</tabstop>
  <tabstop>doubleSpinBox_samplingFrequency</tabstop>
  <tabstop>groupBox_advancedFeatures</tabstop>
  <tabstop>doubleSpinBox_amplitude</tabstop>
  <tabstop>doubleSpinBox_attenuationFactor</tabstop>
  <tabstop>groupBox_visualizationFeatures</tabstop>
  <tabstop>spinBox_dataWindowWidth</tabstop>
  <tabstop>spinBox_dataWindowHeight</tabstop>
  <tabstop>comboBox_lineStyle</tabstop>
  <tabstop>pushButton_color</tabstop>
 </tabstops>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
